### PR TITLE
[Mesheryctl] Meshkit (session__sync_handlers)

### LINKED ISSUE #3305: [Chore] Update server spec, session sync handlers to use new Meshkit Errors
Note: This is a child issue of #3275 
#### Current Behavior
Meshery Server (and its other components, like Adapters) use a centralized library, MeshKit, to provide uniform utilities (e.g. raising and logging errors). Currently, Meshery Server uses a now antiquated style of error logging with logrus.

#### Desired Behavior
Ideally, all use of logrus is replaced with use of MeshKit-style error logging.

#### Implementation
Meshery Handlers needs to be:
 - Updated to use meshkit errors. 
 - Switch to handler Log instead of `logrus`

Handlers to update:

- [ ] server spec handler
- [ ]  session sync handler

For reference, you can checkout  the [`grafana handler`](https://github.com/layer5io/meshery/blob/master/handlers/grafana_handlers.go) and [`prometheus handler`](https://github.com/layer5io/meshery/blob/master/handlers/prometheus_handlers.go) or the [docs](https://docs.meshery.io/project/contributing-error).
Define all Error functions for all the handlers to be written in the `handlers/error.go` file.

#### Mockups

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Chadha93 <gauravchadha1676@gmail.com>

**Description**

This PR fixes #3305 

**Notes for Reviewers**
This pr is for handler 2. `session_sync_handlers`

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/session_sync_handler.go
Comment (sayantan1413) [inline handlers/session_sync_handler.go:64]: Hey @Chadha93 I think I missed this so please consider changing the http.Error to `http.Error(w, ErrMarshal(err, obj).Error(), http.StatusInternalServerError)`

**Review by sayantan1413 — CHANGES_REQUESTED** [review]

**Comment by sayantan1413** [comment]
Hey @mgfeller @DelusionalOptimist I found that in this PR in the ` errorutil_analyze_errors.json` file the path name changed from `"path": "handlers/error.go"` to `"path": "handlers\\error.go"` like all the "/" changed to "\\" so is this a desired behaviour?

**Comment by DelusionalOptimist** [comment]
> Hey @mgfeller @DelusionalOptimist I found that in this PR in the ` errorutil_analyze_errors.json` file the path name changed from `"path": "handlers/error.go"` to `"path": "handlers\\error.go"` like all the "`/`" changed to "`\\`" so is this a desired behaviour?

Hmm.. interesting. If I'm right, this is how paths work in Windows.
@Chadha93 are you doing your changes and running the utility on windows by any chance?

**Comment by sayantan1413** [comment]
Yes @DelusionalOptimist @Chadha93 is running the utility on windows.

**Comment by DelusionalOptimist** [comment]
Ig some reviewer/maintainer can run the utility for Gaurav and update his branch. Or he can push his commits without the three `errorutil_*.json` files for now and we can merge it. Someone else running the util in an open PR will update these. 
First sounds easier tho :P.

**Comment by Chadha93** [comment]
@DelusionalOptimist Yes I tested these changes in windows itself, OK I think these err utilities are not configured for windows ?? I will test it on ubuntu in the future.

**Comment by mgfeller** [comment]
@Chadha93  thanks for testing on Windows, it should work there too, path handling in the code should handle that - not quite sure how it manages command line parameters specifying paths, though, come to think of it, i.e. if they are specified with '/' e.g. in `Makefile` (no paths there yet I can see). Are you using `make`?

`path` is just informative, from that point of view it doesn't matter if it shows a Windows path, but of course it will change if it is run on Linux. I'll look into converting it to a POSIX path in the code.

// @DelusionalOptimist @sayantan1413

**Comment by Chadha93** [comment]
> @Chadha93 thanks for testing on Windows, it should work there too, path handling in the code should handle that - not quite sure how it manages command line parameters specifying paths, though, come to think of it, i.e. if they are specified with '/' e.g. in `Makefile` (no paths there yet I can see). Are you using `make`?
> 
> `path` is just informative, from that point of view it doesn't matter if it shows a Windows path, but of course it will change if it is run on Linux. I'll look into converting it to a POSIX path in the code.
> 
> // @DelusionalOptimist @sayantan1413

@mgfeller Yes I'm using make

**Comment by kumarabd** [comment]
@DelusionalOptimist @sayantan1413 Is this ready to merge?

**Comment by sayantan1413** [comment]
No @kumarabd I think there are some changes still required for this PR to merge.

File: handlers/session_sync_handler.go
Comment (Chadha93) [reply handlers/session_sync_handler.go:64]: @DelusionalOptimist @sayantan1413 should I make changes to this only ( if the windows part is not conflicting).

**Comment by Chadha93** [comment]
@sayantan1413 I think @DelusionalOptimist is right after merging this or someone else running these tests on ubuntu/macOS can update the error utilities. I have updated the seeesion_sync_handlers.

**Comment by sayantan1413** [comment]
Okay @Chadha93 so It looks good to me now. But I think there are some CI issues😬 Can you check it once?

**Comment by mgfeller** [comment]
> Okay @Chadha93 so It looks good to me now. But I think there are some CI issuesgrimacing Can you check it once?

There's an unresolved merge conflict in `handlers/error.go` as well as duplicate declarations. There was a discussion in another PR (?) how to best handle this @sayantan1413 @DelusionalOptimist , related to errors... one possible solution could also be not to declare all errors in the same file in a package (i.e. `error.go`)?

**Comment by Chadha93** [comment]
I have linked a new pr for this (tested on ubuntu), so closing this one.

**Pull request closed without merging**
